src/audio/pipeline.c

```diff
@@ -129,6 +129,8 @@ int pipeline_connect(struct comp_dev *comp, struct comp_buffer *buffer,
 	list_item_prepend(buffer_comp_list(buffer, dir),
 			  comp_buffer_list(comp, dir));
 	buffer_set_comp(buffer, comp, dir);
+	if (!comp->is_shared)
+		dcache_writeback_region(comp, sizeof(*comp));
```

Comment:
What is the exact scenario where it's needed?

Reply:
After commenting this two added line, I can read logs like this:
```
[    929923] (        32) c1 smart_amp-test 1.0   smart_amp_prepare()
[    929936] (        12) c1 smart_amp-test 1.0   ERROR smart_amp_prepare(), source_buffer->source addr= 0xBE074700 comp.type= 0
[    929945] (        10) c1 smart_amp-test 1.0   ERROR smart_amp_prepare(), source_buffer->source addr= 0xBE074000 comp.type= 1
[    929955] (         9) c1 smart_amp-test 1.0   ERROR smart_amp_prepare(), saved in sad sink_buf_addr= 0xBE072180 feedback_buf_addr= 0x0 source_buf_addr= 0xBE072100
```
_comp.type: SOF_COMP_NONE = 0, SOF_COMP_HOST = 1, SOF_COMP_DEMUX ~ 18_
In situation where only smart_amp component is working on another core. This leads to FW crash because of NULL pointer reference in following steps. Feedback_buf is read in such a loop:
```
	/* searching for stream and feedback source buffers */
	list_for_item(blist, &dev->bsource_list) {
		source_buffer = container_of(blist, struct comp_buffer,
					     sink_list);
		buffer_control_invalidate(source_buffer);
+		comp_err(dev, "smart_amp_prepare(), source_buffer->source addr= 0x%X comp.type= %d", (uint32_t)source_buffer->source, source_buffer->source->comp.type);
		if (source_buffer->source->comp.type == SOF_COMP_DEMUX)
			sad->feedback_buf = source_buffer;
		else
			sad->source_buf = source_buffer;
	}
```
So demux component _(working on master core)_ may contain outdated cashed values seen from smart_amp core without this writeback.